paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -887,32 +893,67 @@ def filter_autoscaling_tasks(
     config: MarathonServiceConfig,
+    system_paasta_config: SystemPaastaConfig,
 ) -> Tuple[Mapping[str, MarathonTask], Sequence[Task]]:
+    """Find the tasks that are serving traffic. We care about this because many tasks have a period of high CPU when
+    they first start up, during which they warm up code, load and process data, etc., and we don't want this high load
+    to drag our overall load estimate upwards. Allowing these tasks to count towards overall load could cause a cycle of
+    scaling up, seeing high load due to new warming-up containers, scaling up, until we hit max_instances.
+
+    However, accidentally omitting a task that actually is serving traffic will cause us to underestimate load; this is
+    generally much worse than overestimating, since it can cause us to incorrectly scale down or refuse to scale up when
+    necessary. For this reason, we look at several sources of health information, and if they disagree, assume the task
+    is serving traffic.
+    """
     job_id_prefix = "{}{}".format(
         format_job_id(service=config.service, instance=config.instance),
         MESOS_TASK_SPACER,
     )
 
-    # Get a dict of healthy tasks, we assume tasks with no healthcheck defined
-    # are healthy. We assume tasks with no healthcheck results but a defined
-    # healthcheck to be unhealthy (unless they are "old" in which case we
-    # assume that marathon has screwed up and stopped healthchecking but that
-    # they are healthy
+    # Get a dict of healthy tasks, we assume tasks with no healthcheck defined are healthy.
+    # We assume tasks with no healthcheck results but a defined healthcheck to be unhealthy, unless they are "old" in
+    # which case we assume that Marathon has screwed up and stopped healthchecking but that they are healthy.
+
     log.info("Inspecting %s for autoscaling" % job_id_prefix)
-    marathon_tasks = {}
-    for app in marathon_apps:
-        for task in app.tasks:
-            if task.id.startswith(job_id_prefix) and (
+
+    relevant_tasks_by_app: Dict[MarathonApp, List[MarathonTask]] = {
+        app: app.tasks
+        for app in marathon_apps
+        if app.id.lstrip("/").startswith(job_id_prefix)
+    }
+
+    healthy_marathon_tasks: Dict[str, MarathonTask] = {}
+
+    for app, tasks in relevant_tasks_by_app.items():
+        for task in tasks:
+            if (
                 is_task_healthy(task)
                 or not app.health_checks
                 or is_old_task_missing_healthchecks(task, app)
             ):
-                marathon_tasks[task.id] = task
+                healthy_marathon_tasks[task.id] = task
+
+    service_namespace_config = load_service_namespace_config(
+        service=config.service, namespace=config.get_nerve_namespace()
+    )
+    if service_namespace_config.is_in_smartstack():
+
+        for task in filter_tasks_in_smartstack(
+            [task for tasks in relevant_tasks_by_app.values() for task in tasks],
```

Comment:
If you want to put one more task in here, do the kwarg

```
    tasks=[tasks for ....
```

Reply:
smurfy!